### ISSUE TITLE
Ensure all @TODO comments are needed

### ISSUE THREAD

**Issue by JohnAlbin** [AmazeeLabs/gatsby-starter-amazee#150]
We have a bunch of TODO comments. And many of them are things that developers don't need _to do_ when starting a new project.

We should go through the list and anything that we need to fix in the starter should use FIXME instead.

All the TODO comments in be useful for a new project to read right away.

Additionally, we should change any `@TODO` and `@FIXME` to remove the `@` to make it consistent with the majority of our code comments.

**Comment by pmelab** [comment]
Maybe we have to differentiate between "have to do" and "keep in mind, you might need this later". And I agree with turning our actual starter-kit todo's into `FIXME`s.

**Comment by dan2k3k4** [comment]
If we switch to using `FIXME`, we could setup a pre-push hook that warns about `FIXME`s in the code e.g.
```
git commit -m 'Empty commit' --allow // code has FIXME's outside of this commit
git push
> INFO: There are 13 FIXME's in the codebase.
```

**Comment by pmelab** [comment]
I have mixed feelings about "Warnings" during commit steps. If there is no time to do it immediately, it stays there and from there on the warning will be ignored. TODO-bot spammed us a lot.

Idea: Block merging of a PR as long as there are `FIXME`s *without* a related issue on Github. That way we can just commit and keep pace, but at least don't release undocumented problems. You have the choice to either fix the before merging or at least create a proper followup issue.

**Comment by dan2k3k4** [comment]
Hmm how would we keep track of which `FIXME`'s have issues without using a database? Unless the Github Action workflow reads all open issues and checks if a title (or description) matches the title in the code of the PR...

Anyway, for warnings - yeah at some point everyone just gets used to ignoring them 🙈.

**Comment by pmelab** [comment]
Yes, I would add them to the description:

```
// FIXME: this is ugly (https://github.com/AmazeeLabs/gatsby-starter-amazee/issues/150)
```

**Comment by pmelab** [comment]
Perhaps using that one? https://github.com/pgilad/leasot

**Comment by JohnAlbin** [comment]
> Maybe we have to differentiate between "have to do" and "keep in mind, you might need this later".

Yes, totally agree. `TODO` should mean "have to do". "keep in mind" can just be regular code comments. I've updated the issue description.

**Comment by pmelab** [comment]
A searchable pattern that can be picked up by the various `TODO` IDE integrations. Webstorm for example allows to configure different keywords and color schemes and icons:

<img width="596" alt="Screenshot 2020-07-11 at 09 02 33" src="https://user-images.githubusercontent.com/501415/87218692-4544e180-c355-11ea-9380-2a2ec3ab8dc9.png">

<img width="820" alt="Screenshot 2020-07-11 at 09 01 34" src="https://user-images.githubusercontent.com/501415/87218679-221a3200-c355-11ea-96e8-5b6d5d101c48.png">

I'm hereby suggesting `// FYI:` (despite misspellings in the screenshots)
